docusaurus/docs/create-a-plugin/develop-a-plugin/build-a-streaming-data-source-plugin.md

```diff
@@ -19,106 +19,100 @@ In Grafana, you can set your dashboards to automatically refresh at a certain in
 
-### Step 1: Edit the `plugin.json` file
+1. Enable streaming for your data source in the `plugin.json` file.
 
-Enable streaming for your data source in the `plugin.json` file.
+  ```json
+  {
+    "streaming": true
+  }
+  ```
 
-```json
-{
-  "streaming": true
-}
-```
-
-### Step 2: Change the signature of the `query` method
-
-Modify the signature of the `query` method to return an `Observable` from the `rxjs` package. Make sure you remove the `async` keyword.
+1. Modify the signature of the `query` method to return an `Observable` from the `rxjs` package. Make sure you remove the `async` keyword.
 
-```ts
-import { Observable } from 'rxjs';
-```
+  ```ts
+  import { Observable } from 'rxjs';
+  ```
 
-```ts
-query(options: DataQueryRequest<MyQuery>): Observable<DataQueryResponse> {
-  // ...
-}
-```
+  ```ts
+  query(options: DataQueryRequest<MyQuery>): Observable<DataQueryResponse> {
+    // ...
+  }
+  ```
 
-### Step 3: Create an `Observable` instance for each query
+1. Create an `Observable` instance for each query, and then combine them all using the `merge` function from the `rxjs` package.
 
-Create an `Observable` instance for each query, and then combine them all using the `merge` function from the `rxjs` package.
+  ```ts
+  import { Observable, merge } from 'rxjs';
+  ```
 
-```ts
-import { Observable, merge } from 'rxjs';
-```
-
-```ts
-const observables = options.targets.map((target) => {
-  return new Observable<DataQueryResponse>((subscriber) => {
-    // ...
+  ```ts
+  const observables = options.targets.map((target) => {
+    return new Observable<DataQueryResponse>((subscriber) => {
+      // ...
+    });
   });
-});
-
-return merge(...observables);
-```
 
-### Step 4: Create a `CircularDataFrame` instance
+  return merge(...observables);
+  ```
 
-In the `subscribe` function, create a `CircularDataFrame` instance.
+1. In the `subscribe` function, create a `CircularDataFrame` instance.
 
-```ts
-import { CircularDataFrame } from '@grafana/data';
-```
+  ```ts
+  import { CircularDataFrame } from '@grafana/data';
+  ```
 
-```ts
-const frame = new CircularDataFrame({
-  append: 'tail',
-  capacity: 1000,
-});
-
-frame.refId = query.refId;
-frame.addField({ name: 'time', type: FieldType.time });
-frame.addField({ name: 'value', type: FieldType.number });
-```
+  ```ts
+  const frame = new CircularDataFrame({
+    append: 'tail',
+    capacity: 1000,
+  });
 
-Circular data frames have a limited capacity. When a circular data frame reaches its capacity, the oldest data point is removed.
+  frame.refId = query.refId;
+  frame.addField({ name: 'time', type: FieldType.time });
+  frame.addField({ name: 'value', type: FieldType.number });
+  ```
 
-### Step 5: Send the updated data frame
+  Circular data frames have a limited capacity. When a circular data frame reaches its capacity, the oldest data point is removed.
```

Comment:
```suggestion
Circular data frames have a limited capacity. When a circular data frame reaches its capacity, the oldest data point is removed.
```

Reply:
This whitespace is required so it stays in-line with the rest of the bulleted section, are you suggesting that it should break out of the bulleted list?

Reply:
My mistake. I don't see this extra space often and prefer not to see it. Steps shouldn't put important information after the code block because users easily miss it. (Arguably these sentences should be formatted as a note, but this is a nit.)